src/System.Runtime/tests/System/ArrayTests.cs

```diff
@@ -129,49 +130,239 @@ public static void Construction_MultiDimensionalArray()
             Assert.Equal(array.GetValue(0, 0, 0, 2), 3);
         }
 
-        public static IEnumerable<object[]> BinarySearch_NonGeneric_TestData()
+        public static IEnumerable<object[]> BinarySearch_TestData()
         {
             string[] stringArray = new string[] { null, "aa", "bb", "bb", "cc", "dd", "ee" };
+            string stringDummy = "";
```

Comment:
But this is only an issue because ~10 of the entries you added try to pass null as the value, which forced you to change the parameter to be ```object value```, which meant you needed to add a dummy additional parameter of type T.  If you just got rid of those ~10 entries, then you could make the parameter ```T value```, and not need the additional dummy parameter.  Almost all of those entries don't make sense for the generic version, anyway, as null isn't a valid value for those primitive types, so doing this would also let you remove the special case that bails early in the case of null, remove the now unnecessary casts from object to T, etc.  You can then add just those ~10 entries as data input to the non generic test, where null can make sense.